File: 0287-Find-the-Duplicate-Number.py
class Solution(object):
    def findDuplicate(self, nums):
        slowPointer = nums[0]
        fastPointer = nums[nums[0]]
        
        while slowPointer != fastPointer:
            slowPointer = nums[slowPointer]
            fastPointer = nums[nums[fastPointer]]
            
        fastPointer = 0
        
        while slowPointer != fastPointer:
            slowPointer = nums[slowPointer]
            fastPointer = nums[fastPointer]
            
        return slowPointer

File: 0287-Find-the-Duplicate-Number.js
var findDuplicate = function(nums) {
    let slow = nums[0]
    let fast = nums[nums[0]];
    
    while (slow !== fast) {
        slow = nums[slow];
        fast = nums[nums[fast]];
    }
    
    fast = 0;
    while (slow !== fast) {
        slow = nums[slow];
        fast = nums[fast];
    }
    
    return slow;
};